src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/BackgroundDocumentGenerator.cs

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    [Shared]
+    [Export(typeof(ProjectSnapshotChangeTrigger))]
+    internal class BackgroundDocumentGenerator : ProjectSnapshotChangeTrigger
```

Comment:
Hmm, I think I'd actually take the BackgroundDocumentGenerator from VisualStudio.LanguageServices.Razor and put it into VisualStudio.Editor.Razor. That's the shared binary for both VS platforms. Unless i'm missing something that is. From your changes it looks like you didn't change the core signature of this class (super dope!)